db/src/bootstrap.rs

```diff
@@ -150,6 +151,57 @@ fn idents_to_assertions(idents: &[(&str, i64)]) -> Vec<Value> {
         .collect()
 }
 
+/// Convert {:ident {:key :value ...} ...} to vec![(String(:ident), String(:key), TypedValue(:value)), ...].
+///
+/// Such triples are closer to what the transactor will produce when processing
+/// :db.install/attribute assertions.
+fn symbolic_schema_to_triples(ident_map: &IdentMap, symbolic_schema: &Value) -> Result<Vec<(String, String, TypedValue)>> {
+    // Failure here is a coding error, not a runtime error.
+    let mut triples: Vec<(String, String, TypedValue)> = vec![];
+    // TODO: Consider `flat_map` and `map` rather than loop.
+    match *symbolic_schema {
+        Value::Map(ref m) => {
+            for (ident, mp) in m {
+                let ident = match ident {
+                    &Value::NamespacedKeyword(ref ident) => ident.to_string(),
+                    _ => bail!(ErrorKind::BadBootstrapDefinition(format!("Expected namespaced keyword for ident but got '{:?}'", ident)))
+                };
+                match *mp {
+                    Value::Map(ref mpp) => {
+                        for (attr, value) in mpp {
+                            let attr = match attr {
+                                &Value::NamespacedKeyword(ref attr) => attr.to_string(),
+                                _ => bail!(ErrorKind::BadBootstrapDefinition(format!("Expected namespaced keyword for attr but got '{:?}'", attr)))
+                            };
+
+                            // We have symbolic idents but the transactor handles entids.  Ad-hoc
+                            // convert right here.  This is a fundamental limitation on the
+                            // bootstrap symbolic schema format; we can't represent "real" keywords
+                            // at this time.
+                            //
+                            // TODO: remove this limitation, perhaps by including a type tag in the
+                            // bootstrap symbolic schema, or by representing the initial bootstrap
+                            // schema directly as Rust data.
+                            let typed_value = match TypedValue::from_edn_value(value) {
+                                Some(TypedValue::Keyword(ref s)) => TypedValue::Ref(*ident_map.get(s).ok_or(ErrorKind::UnrecognizedIdent(s.clone()))?),
```

Comment:
I think it could be made neater, but I want to keep the error handling I've handled.  Next time!